src/Expander/Task.hs

```diff
@@ -36,10 +36,11 @@ data ExpanderTask
   | AwaitingDefn Var Ident Binding SplitCorePtr Ty SplitCorePtr Syntax
     -- ^ Waiting on var, binding, and definiens, destination, syntax to expand
   | AwaitingType SplitTypePtr [AfterTypeTask]
-  | ExpandDecl DeclPtr Scope Syntax DeclValidityPtr
-    -- ^ Where to put it, the scope, the decl, where to put the phase
-  | ExpandMoreDecls ModBodyPtr Scope Syntax DeclValidityPtr
-    -- Depends on the binding of the name(s) from the decl and the phase
+  | ExpandModBody ModBodyPtr Scope Syntax DeclValidityPtr
+    -- ^ The tree node to fill, the scope, the decls, and how to notify that all
+    -- the names have been bound.
+  | ExpandDependentModBody ModBodyPtr Scope Syntax DeclValidityPtr
+    -- ^ Depends on the binding of the names in DeclValidityPtr
```

Comment:
`ExpandDecl` and `ExpandMoreDecls` were expanding the head and the tail of a list of declarations. The tail is blocked until the head's name is bound. Now, `ExpandModBody` is expending a node in the tree, while `ExpandDependentModBody` is expanding the right child of a node in the tree, by blocking until some name is bound. The intent is to wait until all the names in the left child of the node are bound, but I haven't tried to make sure that's the case yet.

The comments for `ExpandDecl` and `ExpandMoreDecls` didn't make 100% sense to me, so I hope I didn't lose anything important when I rewrote the comments to describe `ExpandModBody` and `ExpandDependentModBody` instead.

Reply:
(now named `ExpandDeclTree` and `ExpandDependentDeclTree`)

Reply:
Sounds good!